plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,279 @@
+            -- menu entry: filename dialog
+            {
+                text = _("Set system screensaver image"),
+                checked_func = function()
+                    return self.cover_image_path ~= "" and pathOk(self.cover_image_path)
+                end,
+                help_text = _("The cover of the current book will be stored in this file."),
+                keep_menu_open = true,
+                callback = function(menu)
+                    local InputDialog = require("ui/widget/inputdialog")
+                    local sample_input
+                    sample_input = InputDialog:new{
+                        title = _("Screensaver image filename"),
+                        input = self.cover_image_path,
+                        input_type = "string",
+                        description = _("You can enter the filename of the cover image here."),
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(sample_input)
+                                    end,
+                                },
+                                {
+                                    text = _("Save"),
+                                    is_enter_default = true,
+                                    callback = function()
+                                        local new_cover_image_path = sample_input:getInputText()
+                                        if new_cover_image_path ~= self.cover_image_path then
+                                            self:cleanUpImage() -- with old filename
+                                            self.cover_image_path = new_cover_image_path -- update filename
+                                            G_reader_settings:saveSetting("cover_image_path", self.cover_image_path)
+                                            local is_path_ok, is_path_ok_message = pathOk(self.cover_image_path)
+                                            if self.cover_image_path ~= "" and is_path_ok then
+                                                self:createCoverImage(self.ui.doc_settings) -- with new filename
+                                            else
+                                                self.enabled = false
+                                                UIManager:show(InfoMessage:new{
+                                                    text = is_path_ok_message,
+                                                    show_icon = true,
+                                                })
+                                            end
+                                        end
+                                        UIManager:close(sample_input)
+                                        menu:updateItems()
+                                    end,
+                                },
```

Comment:
Wouldn't it be better to check if the path is correct, and if it is only then do the clean up? Right now if user accidentally put in wrong path they can loose they cover images, right?

Reply:
Guess not, since entering invalid path is meant to disable the plugin.